src/axe-report.ts

```diff
@@ -38,14 +38,33 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
     });
   });
 */
+import * as axeTypes from 'axe-core';
 
-export async function axeReport(dom, config = {}) {
+declare global {
+  namespace axe {
+    const run: typeof axeTypes.run;
+  }
+}
+
+export interface AxeReportOptions extends axeTypes.RunOptions {
+  cleanup?: () => Promise<void>
+  axeConfig?: axeTypes.RunOptions
+}
+
+interface AxeReportRunOptions extends axeTypes.RunOptions {
+  resultTypes: string[]
+}
+
+export async function axeReport(dom: axeTypes.ElementContext, config:AxeReportOptions = {}) {
   const {cleanup, axeConfig} = config;
   const {violations} = await axe.run(dom, axeConfig || {
-    runOnly: ['wcag2a', 'wcag2aa', 'section508'],
+    runOnly: {
+      type: 'tag',
+      values: ['wcag2a', 'wcag2aa', 'section508']
+    },
     // Ignore tests that are passing.
     resultTypes: ['violations']
-  });
+  } as AxeReportRunOptions);
```

Comment:
`as` should be used as little as possible, as it's more powerful than you usually mean. One way to write this:

```typescript
// At the top level
const defaultConfig: AxeReportRunOptions = {
  runOnly: {
    type: 'tag',
    values: ['wcag2a', 'wcag2aa', 'section508']
  },
  // Ignore tests that are passing.
  resultTypes: ['violations']
};

// then in axeReport
const {violations} = await axe.run(dom, axeConfig || defaultConfig);
```